src/main/java/com/googlecode/lanterna/gui2/AbstractBasePane.java

```diff
@@ -44,6 +45,8 @@
     private boolean strictFocusChange;
     private boolean enableDirectionBasedMovements;
     private Theme theme;
+    
+    private Interactable mouseDownForDrag = null;
```

Comment:
This makes it so that the component which experienced the mouse down is the component which the mouse drags get routed to.